### PR TITLE
feat: allow connecting to instances that set CSP and x-frame-options

### PR DESCRIPTION
**Pull request by csett86**
While in browser environments the headers are sensible, the only purpose
of the electron app is load jitsi in the iframe api. Thus the security
implications are accepted to work with more jitsi instances.

Fixes: #285

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -228,6 +228,17 @@ function createJitsiMeetWindow() {
 
     mainWindow.webContents.setWindowOpenHandler(windowOpenHandler);
 
+    // Filter out x-frame-options and CSP to allow loading jitsi via the iframe API
+    // Resolves https://github.com/jitsi/jitsi-meet-electron/issues/285
+    mainWindow.webContents.session.webRequest.onHeadersReceived((details, callback) => {
+        delete details.responseHeaders['x-frame-options'];
+        delete details.responseHeaders['content-security-policy'];
```
Comment (saghul) [inline main.js:235]: I thought we had established that some CSPs do work?

**Review by saghul — DISMISSED** [review]
The change LGTM (I did add a question) but I find the principle troublesome.

If a server admin sets X-Frame-Options they are being explicit about not wanting to be embedded. Browsers fulfill their wishes, so why are we ignoring them?

**Comment by csett86** [comment]
My reasons are:

1. From a users perspective the app does not embed the conference, but loads it fullscreen and does not provide any additional functionality.

2. This would bring the mobile apps and the electron app in sync, as the mobile app also ignores these headers and loads conferences from servers that send these headers.

So for me the electron app should behave like the mobile apps and not like a browser here.

To reproduce: A public example of a server which sets those headers is https://sichere-videokonferenz.de/

**Comment by saghul** [comment]
Makes sense, the mobile example is a good point.

Let's give it a go!

File: main.js (hunk 1 above)
Comment (csett86) [reply main.js]: Good point, I changed the approach to only filter out the frame-ancestors CSP and keep everything else in place. With that, CSP would eg. still protect against loading 3rd party resources in the iframe itself if this is enforced via CSP.

**Comment by csett86** [comment]
As I updated the handling of the CSP, could you please reapprove, @saghul?

**Review by saghul — APPROVED** [review]

**Pull request merged**